### ISSUE TITLE
uc-crux-llvm: False positives on skipped functions like `getline`

### ISSUE THREAD

**Issue by langston-barrett** [GaloisInc/crucible#907]
Consider this program:
```c
#include <stdio.h>

void do_getline(FILE *f) {
  char loc[256];
  char *line = NULL;
  size_t linesize = 0;
  while (getline(&line, &linesize, f) != -1) {
    printf("%c", line[0]);
  }
}
```
UC-Crux-LLVM thinks there's probably a bug here:
```
[UC-Crux-LLVM] Read from data that concretely wasn't a pointer at do_getline.c:8:5
```
Why? Well, `getline` is a POSIX function. From `getline(3)`:
```
NAME
       getline, getdelim - delimited string input

SYNOPSIS
       #include <stdio.h>
       ssize_t getline(char **restrict lineptr, size_t *restrict n, FILE *restrict stream);

[...]
DESCRIPTION
       getline()  reads  an entire line from stream, storing the address of the
       buffer containing the text into *lineptr.  The buffer is null-terminated
       and includes the newline character, if one was found.

       If *lineptr is set to NULL before the call, then getline() will allocate
       a buffer for storing the line.  This buffer should be freed by the  user
       program even if getline() failed.
```
But UC-Crux doesn't have an override for it, so it just gets skipped by default. This leaves `line` with its original value (`NULL`), leading to a concrete null pointer dereference.

The fix is that the heuristic involving concretely-non-pointer dereferences should be disabled if any functions were skipped during execution.

**Comment by langston-barrett** [comment]
The mechanism that fixed #982 (documented [here](https://github.com/GaloisInc/crucible/blob/555acf77477a02c8b0e101c2c5fcdb9a88338382/uc-crux-llvm/doc/specs.md)) may (eventually, with more work) be useful in such situations: the user could provide specs describing how `getline` writes to its arguments.

**Comment by langston-barrett** [comment]
With #1003, you can provide a spec for such a function. With #1036, UC-Crux will warn you that the result may be a false positive. I think we can consider this fixed.
